### DISCUSSION TITLE
Random Error: Cannot destructure property 'gasLimit' of '(intermediate value)' as it is null.

### DISCUSSION THREAD

**Discussion by karacurt** [wevm/wagmi#3701] [Q&A] [answered]
Once in a while my writeContract transaction return the following error: 

`Cannot destructure property 'gasLimit' of '(intermediate value)' as it is null. `

Someone knows what i can i do to prevent this to happen?

**Comment by alyavasilyeva** [comment]
We've got the same issue, happens pretty randomly =\

**Reply by JavierFerrerGomez** [reply]
Hi, I am having the same problem. Seems like adding the **gas** property solved the issue for me  (hard to tell 100% as it was pretty random)
  ```
  const { config: playWriteConfig } = usePrepareContractWrite({
        address: theAddress,
        abi: theAbi,
        functionName: 'play',
        args: [arg1, arg2],
        value: totalPlayValue,
        gas: 1000000, <------ I just added this
    } as any)
    const { data: writeData, write: writePlay } = useContractWrite({
        ...playWriteConfig,
        onSuccess() {
            // blah
        },
    })
```

I am using wagmi 1.2

**Reply by chulavege1** [reply]
@JavierFerrerGomez Thanks, new gas prop fix my problem.
But no any info about it in wagmi docs.

**Comment by karacurt** [comment, marked as answer]
Guys i was able to solve the problem with those package versions:

```json
"next": "14.0.4",
"typescript": "5.1.6",
"viem": "2.7.6",
"wagmi": "2.5.6",
"@tanstack/react-query": "^5.26.3",
"@web3modal/wagmi": "^4.1.1"
```

And also changing the import of the wagmi config to get it from the web3modal package:

```ts
import { defaultWagmiConfig } from '@web3modal/wagmi/react/config'

import { polygon } from 'wagmi/chains'

export const projectId = process.env.WALLET_CONNECT_PROJECT_ID

const metadata = {
  name: 'Orium Marketplace',
  description: 'by Orium Network',
  url: 'https://orium.network',
  icons: ['https://cdn.orium.network/images/misc/logo.svg'],
}

// Create wagmiConfig
const chains = [polygon] as const
export const config = defaultWagmiConfig({
  chains,
  projectId,
  metadata,
  enableCoinbase: false,
})
```

I used this repository from web3modal as reference: https://github.com/WalletConnect/web3modal/tree/V4/examples/next-wagmi

**Reply by pociej** [reply]
How you can be sure this solves the problem? Which version of which package is the source of it?
